lkmpg.tex

```diff
@@ -793,17 +801,19 @@ \subsection{The proc\_ops Structure}
 It will help with compatibility:
 
 \begin{code}
-struct proc_ops fops = {
-	.proc_read = device_read,
-	.proc_write = device_write,
-	.proc_open = device_open,
-	.proc_release = device_release
+struct file_operations fops = {
+	.read = device_read,
+	.write = device_write,
+	.open = device_open,
+	.release = device_release
 };
 \end{code}
 
 The meaning is clear, and you should be aware that any member of the structure which you do not explicitly assign will be initialized to NULL by gcc.
 
-An instance of struct proc\_ops containing pointers to functions that are used to implement read, write, open, \ldots{} syscalls is commonly named fops.
+An instance of struct \verb|file_operations| containing pointers to functions that are used to implement read, write, open, \ldots{} syscalls is commonly named fops.
+
+In Linux v5.6+, this structure has been instead by \ref{sec:proc_ops} in the \textbf{/proc} file system.
```

Comment:
Change to the following:
> Since Linux v5.6, the `proc_ops` structure was introduced to replace the use of of the `file_operations` structure when registering proc handlers.